src/asm/lexer.cpp

```diff
@@ -1106,28 +1086,137 @@ static uint32_t readFractionalPart(uint32_t integer) {
 }
 
 char binDigits[2];
+char gfxDigits[4];
+
+static bool isValidDigit(char c) {
+	return (c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || c == '.'
+	       || c == '#' || c == '@';
+}
+
+static bool checkDigitErrors(char const *digits, size_t n, char const *type) {
+	for (size_t i = 0; i < n; i++) {
+		char c = digits[i];
+
+		if (!isValidDigit(c)) {
+			error("Invalid digit for %s constant %s\n", type, printChar(c));
+			return false;
+		}
+
+		if (c >= '0' && c < n + '0' && c != i + '0') {
+			error("Changed digit for %s constant %s\n", type, printChar(c));
+			return false;
+		}
```

Comment:
Hmm, I dunno: if someone were to set an inverted palette, then it seems to me that it would make sense to `-g 3210`?

Reply:
The core question is “do we want the default mappings to *always* be available”. It's a good idea, I think? I'll request some different documentation wording to make that more explicit.